frontend/src/metabase/query_builder/containers/use-register-query-builder-metabot-context.ts

```diff
@@ -0,0 +1,18 @@
+import { useRegisterMetabotContextProvider } from "metabase/metabot";
+
+import { getQuestion } from "../selectors";
+
+export const useRegisterQueryBuilderMetabotContext = () => {
+  useRegisterMetabotContextProvider((state) => {
+    const question = getQuestion(state);
+    if (!question) {
+      return {};
+    }
+
+    return {
+      user_is_viewing: question.isSaved()
```

Comment:
Same here @Somtom. Should all logic regarding `user_is_viewing` be removed from FE?

Reply:
I haven't heard of us wanting to remove it. Is it not used for the embedding use case?

Reply:
cc @retro 

We are currently not using this anymore on the ai-service side because we trimmed a few things to focus on getting the embedding use case to work. If it feels cleaner, we can remove it (at least from the FE for now?) but it is very likely that we tackle the internal use case soon which might reuse those patterns. 

We used this `user_is_viewing` initially when the Metabot also had other tools. It could for example filter the results of an ad-hoc query or report (i.e. build on top of it as if it was a data source). However, our query tools changed and we don't support this anymore right now so I am not sure how useful it is to keep. We might do something similar in the future but I don't know how this is going to look like.

Maybe @retro has some thoughts here as well.

Reply:
Well, if there's a possibility to reuse it later on. I wouldn't bother removing it now. I just thought if we know we're not going to use it again, removing it could be better.

Reply:
> I haven't heard of us wanting to remove it. Is it not used for the embedding use case?

@metamben It's not used in Embedding. Or rather, we just didn't do anything special to remove `user_is_viewing` as far as I know. I chatted with @Somtom around the embedding offsite week (24-28 Mar) that this property isn't used anymore.

Reply:
@Somtom @WiNloSt So, for the embedding use case, I would assume this is always empty? I wouldn't remove it if it's not necessary because this allows us to continue working on the internal use case. 

If this is causing issues for the embedding use case, that should take precedence, and in that case we can remove it

Reply:
Embedding doesn't use it, whether it's empty or has value doesn't change the behavior whatsoever. I just pointed out the fact that it's currently not used. If there's an intent to keep it then let them be :)